### ISSUE TITLE
[Gateway] Version 7.0.0 Don't login correctly with oauth2

### ISSUE THREAD

**Issue by Adri-Glez** [jhipster/generator-jhipster#14445]
##### **Overview of the issue**

I create a empty gateway project with the last release of Jhipster 7.0.0 with oauth2 keycloak authorization. It starts without errors but the restricted/secured area don’t works. I login to the admin user but after that the browser redirect to  this url  http://localhost:8098/login?error with this json error:
```javascript
{
  "type" : "https://www.jhipster.tech/problem/problem-with-message",
  "title" : "Unauthorized",
  "status" : 401,
  "detail" : "Not Authenticated",
  "path" : "/login",
  "message" : "error.http.401"
}
```
In the console only appears this warn messages:
```
2021-03-24 19:18:02.075  WARN 19844 --- [oundedElastic-3] o.z.problem.spring.common.AdviceTraits   : Unauthorized: Not Authenticated
2021-03-24 19:18:03.444  WARN 19844 --- [oundedElastic-3] o.z.problem.spring.common.AdviceTraits   : Unauthorized: Not Authenticated
2021-03-24 19:18:59.512 DEBUG 19844 --- [ctor-http-nio-2] jdk.event.security                       : X509Certificate: Alg:SHA256withRSA, Serial:1785d457e52, Subject:CN=jhipster, Issuer:CN=jhipster, Key type:RSA, Length:2048, Cert Id:-607563525, Valid from:23/3/21 5:07, Valid until:23/3/31 5:08
2021-03-24 19:18:59.631  WARN 19844 --- [ctor-http-nio-4] o.z.problem.spring.common.AdviceTraits   : Unauthorized: Not Authenticated
```
##### **Motivation for or Use Case**
This bug appears just after create one gateway project with oauth2 before start the development, anyone who want to start with the basic template will start with something that don't works completely. It worked in previous version 6

##### **Reproduce the error**

is easy:
1) create with Jhipster 7.0.0 a new gateway maven project with the oauth2 authenticationType  (you can check the exact configuration in the yo-rc.json section). 
2) run the keycloak in a docker container `docker-compose  -f src\main\docker\app.yml up -d keycloak`
3) run jhipster-registry executing mvnw in a folder of with the jhipster-registry sorce code version 6.7.1. Although it could be launched with the docker compose command of the new project
4) launch the new project of step 1 with mvnw command.

##### **Related issues**

I didn't found.

##### **Suggest a Fix**

I can't yet

##### **JHipster Version(s)**

Jhipster 7.0.0

##### **JHipster configuration**

This is the output of jhipster info:

INFO! Using JHipster version installed locally in current project's node_modules
Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
gateway-web@0.0.1-SNAPSHOT K:\jee_ecli_2020\jhi-gateway-web
`-- generator-jhipster@7.0.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "gateway",
    "baseName": "gatewayWeb",
    "jhipsterVersion": "7.0.0",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": true,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cypress", "gatling", "cucumber"],
    "pages": [],
    "creationTimestamp": 1616514640129,
    "serviceDiscoveryType": "eureka",
    "reactive": true,
    "authenticationType": "oauth2",
    "packageName": "es.adrian",
    "serverPort": "8098",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "packageFolder": "es/adrian",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["en", "es"]
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

java version "14.0.2" 2020-07-14
Java(TM) SE Runtime Environment (build 14.0.2+12-46)
Java HotSpot(TM) 64-Bit Server VM (build 14.0.2+12-46, mixed mode, sharing)

git version 2.17.1.windows.2

node: v15.4.0

npm: 7.6.1

Docker version 20.10.2, build 2291f61

docker-compose version 1.27.4, build 40524192

No change to package.json was detected. No package manager install will be executed.
Congratulations, JHipster execution is complete!

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

No Entities created yet

##### **Browsers and Operating System**

Windows 10 in all Browsers

- [X ] Checking this box is mandatory (this is just to show you read everything)

**Comment by mraible** [comment]
I deleted all my local Docker containers and created a gateway with your config. After creating the app, I launched Keycloak and JHipster Registry with [Oh My Zsh shortcuts](https://www.jhipster.tech/oh-my-zsh/):

```
jhkeycloakup
# Pulling keycloak (jboss/keycloak:12.0.4)...
# wait 10 seconds after it starts
jhregistryup
# Pulling jhipster-registry (jhipster/jhipster-registry:v6.7.1)...
./mvnw
```

I'm able to log in and access the admin sections.

<img width="1141" alt="Screen Shot 2021-03-24 at 9 46 12 PM" src="https://user-images.githubusercontent.com/17892/112415952-646c7f00-8cea-11eb-912f-ecafd9756151.png">

`npm run e2e` passes too:

```
       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  administration/administration.spec.      00:06        5        5        -        -        - │
  │    ts                                                                                          │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:06        5        5        -        -        -

Execution time: 17 s.
```

I've found that if you want to run the JHipster Registry locally with oauth2 enabled (instead of using Docker Compose), you have to use a command like the following:

```
SPRING_PROFILES_ACTIVE=dev,oauth2,native,swagger ./mvnw
```

I like to set it as an alias:

```
alias registry-run="SPRING_PROFILES_ACTIVE=dev,oauth2,native,swagger ./mvnw"
```

**Comment by Adri-Glez** [comment]
Thanks Matt,

I tried launching the register with Docker and locally with the environments settings that you suggest.  Thanks to it, I get  a error about the expired certificate with more information than with my gateway application, I found that the dateTime of the machines in the Windows Docker was wrong and different that the windows host, I update Docker and change some configurations, after that It worked.

The settings and environment variables needed for launching The Jhispter Register and  also the Jhipster Zsh shortcuts that you suggest are useful for knowing better how the launch easly a jhipster aplication but Unfortunately It don’t works in Windows, although It is similar. I create a ‘cmd’ script in the Register folder named ‘LaunchRegister.cmd’ with this code for launch it in Windows:

```
set SPRING_PROFILES_ACTIVE=dev,api-docs,oauth2,native,swagger
set SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI=http://keycloak:9080/auth/realms/jhipster
set SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID=jhipster-registry
set SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET=jhipster-registry
./mvnw -P dev,swagger -ntp -DskipTests
REM ./mvnw -P prod,swagger -ntp -DskipTests
echo Press any key to Close.
pause > nul
```

Maybe is useful to put this script in the source code repository  of jHipster-Register for helping to launch it by the Windows users or explaining something in the documentation for the windows users. Anyway if somebody read this issue comment could be helpful for they.
